File: Fantasy.Base/ServiceModel/LogConsoleService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.ServiceModel;

namespace Fantasy.ServiceModel
{
    public class LogConsoleService : ServiceBase, ILogListener
    {
        public override void InitializeService()
        {
            ILogger logger = (ILogger)((IServiceProvider)this.Site).GetService(typeof(ILogger));
            logger.AddListener(this);
            base.InitializeService();
        }

        public override void UninitializeService()
        {
            base.UninitializeService();
            ILogger logger = (ILogger)((IServiceProvider)this.Site).GetService(typeof(ILogger));
            logger.RemoveListener(this);
        }

        #region IJobLoggerListener Members

        //public void OnStatusChanged(string status)
        //{
        //    Debug.WriteLine("Status: " + status); 
        //}

        public void OnMessage(string category, MessageImportance importance, string message)
        {
            Console.WriteLine("Message: " + message);
        }

        public void OnWaring(string category, Exception exception, MessageImportance importance, string message)
        {
            Console.WriteLine("Warning: " + message);
        }

        public void OnError(string category, Exception exception, string message)
        {
            Console.WriteLine("Error: " + message);
        }

        #endregion
    }
}

File: AutoConverted/Fantasy.Base/ServiceModel/LogConsoleService.java
package Fantasy.ServiceModel;

import Fantasy.ServiceModel.*;

public class LogConsoleService extends AbstractService implements ILogListener
{
	@Override
	public void InitializeService()
	{
		ILogger logger = (ILogger)((IServiceProvider)this.Site).GetService(ILogger.class);
		logger.AddListener(this);
		super.InitializeService();
	}

	@Override
	public void UninitializeService()
	{
		super.UninitializeService();
		ILogger logger = (ILogger)((IServiceProvider)this.Site).GetService(ILogger.class);
		logger.RemoveListener(this);
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IJobLoggerListener Members

	//public void OnStatusChanged(string status)
	//{
	//    Debug.WriteLine("Status: " + status); 
	//}

	public final void OnMessage(String category, MessageImportance importance, String message)
	{
		System.out.println("Message: " + message);
	}

	public final void OnWaring(String category, RuntimeException exception, MessageImportance importance, String message)
	{
		System.out.println("Warning: " + message);
	}

	public final void OnError(String category, RuntimeException exception, String message)
	{
		System.out.println("Error: " + message);
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}